systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/controllers/DeploymentUtils.java

```diff
@@ -240,4 +253,20 @@ public static void waitForNoRollingUpdate(String deploymentName, Map<String, Str
 
         return depConfigSnapshot(name);
     }
+
+    public static <T extends Deployment, L extends DeploymentStatus> void logCurrentDeploymentStatus(T deployment, L status) {
```

Comment:
You don't need to have this generic I think. Or maybe update it to accept resources which has status? In that case you should change the naming.